lua/effects/eff_term_goodarc.lua

```diff
@@ -0,0 +1,484 @@
+
+        end
+
+        self.Points[#self.Points + 1] = midPoint
+
+        local seg2Count = self.Segments - seg1Count
+        local dir2 = ( endPos - midPoint ):GetNormalized()
+        local len2 = ( endPos - midPoint ):Length()
+        local ang2 = dir2:Angle()
+        local right2, up2 = ang2:Right(), ang2:Up()
+
+        for i = 1, seg2Count - 1 do
+            local frac = i / seg2Count
+            local basePos = midPoint + dir2 * len2 * frac
+            local falloff = math.sin( frac * math.pi )
+            local offsetX = math.Rand( -1, 1 ) * self.Jitter * falloff * 0.7
+            local offsetY = math.Rand( -1, 1 ) * self.Jitter * falloff * 0.7
+            self.Points[#self.Points + 1] = basePos + right2 * offsetX + up2 * offsetY
+
+        end
+    else
+        for i = 1, self.Segments - 1 do
+            local frac = i / self.Segments
+            local basePos = startPos + direction * length * frac
+            local falloff = math.sin( frac * math.pi )
+            local offsetX = math.Rand( -1, 1 ) * self.Jitter * falloff
+            local offsetY = math.Rand( -1, 1 ) * self.Jitter * falloff
+            self.Points[#self.Points + 1] = basePos + right * offsetX + up * offsetY
+
+        end
+    end
+
+    self.Points[#self.Points + 1] = endPos
+
+    if self.DoBranches and #self.Points >= 4 then
+        self:GenerateBranches( direction, length )
+
+    else
+        self.Branches = nil
+
+    end
+end
+
+function EFFECT:GenerateBranches( mainDir, mainLength )
+    self.Branches = {}
+
+    local branchCount = self.BranchCount or math.max( 1, math.floor( self.Segments / 2.5 ) )
+    local usedIndices = {}
+
+    for _ = 1, branchCount do
+        local attempts = 0
+        local pointIdx
+
+        repeat
+            pointIdx = math.random( 2, math.max( 2, #self.Points - 2 ) )
+            attempts = attempts + 1
+        until not usedIndices[pointIdx] or attempts > 10
+
+        usedIndices[pointIdx] = true
+
+        local branchStart = self.Points[pointIdx]
+        local branchDir = ( VectorRand() + mainDir * 0.2 ):GetNormalized()
+        local branchLen = mainLength * math.Rand( 0.15, 0.4 )
+        local branchSegs = math.random( 2, 4 )
+
+        local branch = { branchStart }
+        local bAng = branchDir:Angle()
+        local bRight, bUp = bAng:Right(), bAng:Up()
+
+        for j = 1, branchSegs do
+            local t = j / branchSegs
+            local pos = branchStart + branchDir * branchLen * t
+            local jitter = self.Jitter * 0.5 * ( 1 - t * 0.7 )
+            pos = pos + bRight * math.Rand( -1, 1 ) * jitter + bUp * math.Rand( -1, 1 ) * jitter
+            branch[#branch + 1] = pos
+
+        end
+
+        branch.width = math.Rand( 0.4, 0.7 )
+        self.Branches[#self.Branches + 1] = branch
+
+        if branchSegs >= 3 and math.random() > 0.5 then
+            local subStart = branch[2]
+            local subDir = ( VectorRand() + branchDir * 0.1 ):GetNormalized()
```

Comment:
```suggestion
            local subDir = VectorRand() + branchDir * 0.1
            subDir:Normalize()
```
Creates less vector objects